src/flows/service_functions/communication.py

```diff
@@ -69,10 +70,10 @@ def send_message(
         if target_state_obj is not None:
             resolved_mapping.setdefault("target", target_state_obj)
 
-    receiver = cast(
-        Any, target_state or flow_execution.resolve_flow_reference(recipient)
-    )
-    caller_state = cast(Optional[BaseState], resolved_mapping.get("caller"))
+    receiver = target_state or flow_execution.resolve_flow_reference(recipient)
+    caller_state = resolved_mapping.get("caller")
+    if caller_state is not None and not isinstance(caller_state, BaseState):
+        caller_state = None
```

Comment:
Lines 74-76 feel totally redundant because 60-64 already set it as an optional BaseState